tools/pot/openvino/tools/pot/data_loaders/synthetic_image_loader.py

```diff
@@ -0,0 +1,327 @@
+        self._cpu_count = min(os.cpu_count(), self.subset_size)
+        self._shape = config.get('shape', None)
+        self.data_source = config.get('data_source', None)
+        self._weights = np.array([
+            0.2, 1, 1, 1, 1, 1,
+            0.6, 1, 1, 1, 1, 1,
+            1, 1, 1, 1, 1, 1,
+            1.4, 1, 1, 1, 1, 1,
+            1.8, 1, 1, 1, 1, 1,
+            1, 0.2, 1, 1, 1, 1,
+            1, 0.6, 1, 1, 1, 1,
+            1, 1.4, 1, 1, 1, 1,
+            1, 1.8, 1, 1, 1, 1,
+            1, 1, 0.2, 1, 1, 1,
+            1, 1, 0.6, 1, 1, 1,
+            1, 1, 1.4, 1, 1, 1,
+            1, 1, 1.8, 1, 1, 1,
+            1, 1, 1, 0.2, 1, 1,
+            1, 1, 1, 0.6, 1, 1,
+            1, 1, 1, 1.4, 1, 1,
+            1, 1, 1, 1.8, 1, 1,
+            1, 1, 1, 1, 0.2, 1,
+            1, 1, 1, 1, 0.6, 1,
+            1, 1, 1, 1, 1.4, 1,
+            1, 1, 1, 1, 1.8, 1,
+            1, 1, 1, 1, 1, 0.2,
+            1, 1, 1, 1, 1, 0.6,
+            1, 1, 1, 1, 1, 1.4,
+            1, 1, 1, 1, 1, 1.8,
+        ]).reshape(-1, 6)
+        self._threshold = 0.2
+        self._iterations = 200000
+        self._num_of_points = None
+        self._instances = None
+        self._categories = None
+        if isinstance(self._shape, str):
+            self._shape = list(map(int, re.findall(r'\d+', self._shape)))
+
+        super().get_layout()
+        self._check_input_shape()
+
+        if os.path.exists(self.data_source) and os.listdir(self.data_source) and not config.generate_data:
+            logger.info(f'Dataset was found in `{self.data_source}`')
+        else:
+            logger.info(f'Synthetic dataset will be stored in `{self.data_source}`')
+            if not os.path.exists(self.data_source):
+                os.mkdir(self.data_source)
+
+        assert os.path.isdir(self.data_source)
+        if config.generate_data or not os.listdir(self.data_source):
+            self._download_colorization_model()
+            logger.info(f'Start generating {self.subset_size} synthetic images')
+            self.generate_dataset()
+
+        self._img_files = collect_img_files(self.data_source)
+
+    def _check_input_shape(self):
+        if self._shape is None:
+            raise ValueError('Input shape should be specified. Please, use `--shape`')
+        if  len(self._shape) < 3 or len(self._shape) > 4:
+            raise ValueError(f'Input shape should have 3 or 4 dimensions, but provided {self._shape}')
+        if self._shape[self._layout.get_index_by_name('C')] != 3:
+            raise ValueError('SyntheticImageLoader can generate images with only channels == 3')
+
+    def _download_colorization_model(self):
+        proto_name = 'colorization_deploy_v2.prototxt'
+        model_name = 'colorization_release_v2.caffemodel'
+        npy_name = 'pts_in_hull.npy'
+
+        if not os.path.exists(proto_name):
+            url = 'https://raw.githubusercontent.com/richzhang/colorization/caffe/colorization/models/'
```

Comment:
Can we do something to avoid model unavailability in the future? @AlexKoff88 @l-bat

Reply:
The only thing we can do is to train our own model. But this is quite an expensive task because it requires some R&D, Legal clearance, etc.